subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -503,14 +527,13 @@ int bt_mesh_adv_enable(void)
 
 int bt_mesh_adv_disable(void)
 {
-	int err;
 	struct k_work_sync sync;
+	int err;
 
 	for (int i = 0; i < ARRAY_SIZE(advs); i++) {
 		atomic_set_bit(advs[i].flags, ADV_FLAG_SUSPENDING);
 
-		if (k_current_get() != &k_sys_work_q.thread ||
-		    (k_work_busy_get(&advs[i].work) & K_WORK_RUNNING) == 0) {
+		if (k_current_get() != k_work_queue_thread_get(MESH_WORKQ)) {
```

Comment:
Why did you remove `(k_work_busy_get(&advs[i].work) & K_WORK_RUNNING) == 0)` check? It is needed to ensure that we are not calling `k_work_flush` from the advertiser's k_work handler. Otherwise it will deadlock.